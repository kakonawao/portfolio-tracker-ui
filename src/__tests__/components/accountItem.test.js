import React from "react";
import { shallow } from "enzyme";

import AccountItem from "../../components/accountItem";


describe("<AccountItem>", () => {
    it("renders account data with holder", () => {
        const props = {
            key: "1",
            code: "1",
            description: "bank account",
            holder: {
                name: "Bank of Ireland"
            }
        };

        const wrapper = shallow(<AccountItem {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it("renders account data without holder", () => {
        const props = {
            key: "2",
            code: "2",
            description: "cash account",
            holder: null
        };

        const wrapper = shallow(<AccountItem {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
