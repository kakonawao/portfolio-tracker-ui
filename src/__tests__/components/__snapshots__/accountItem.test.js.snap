// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountItem> renders account data with holder 1`] = `
<div>
  <p>
    1
    : 
    bank account
     (
    Bank of Ireland
    )
  </p>
</div>
`;

exports[`<AccountItem> renders account data without holder 1`] = `
<div>
  <p>
    2
    : 
    cash account
     (
    N/A
    )
  </p>
</div>
`;
